extensions/awork/src/tasks.tsx

```diff
@@ -0,0 +1,117 @@
+      />
+      <Action
+        icon={Icon.Clock}
+        title="Log Time"
+        shortcut={{ modifiers: ["ctrl", "cmd"], key: "enter" }}
+        onAction={async () => {
+          try {
+            await launchCommand({
+              name: "logTime",
+              type: LaunchType.UserInitiated,
+              context: {
+                taskId: props.taskId,
+                projectId: props.projectId,
+                typeOfWorkId: props.typeOfWorkId,
+              },
+            });
+          } catch (error) {
+            showFailureToast("Failed to launch time logging", error as Error);
+          }
+        }}
+      />
+    </ActionPanel>
+  );
+};
+
+const TaskItem = (props: { task: task }) => {
+  return (
+    <List.Item
+      title={props.task.name}
+      subtitle={props.task.project.name}
+      keywords={[props.task.project.name, props.task.id]}
+      actions={
+        <Actions taskId={props.task.id} projectId={props.task.projectId} typeOfWorkId={props.task.typeOfWorkId} />
+      }
+    />
+  );
+};
+
+export default function Command(props: LaunchProps) {
+  const [searchText, setSearchText] = useState<string>("");
+  const [projectId, setProjectId] = useState<string>("");
+  const {
+    data: tasks,
+    pagination,
+    isLoading: isLoadingTasks,
+    revalidate: updateTasks,
+  } = useCachedPromise(getTasks, [searchText, 100, projectId], {
+    onData: (data) => {
+      if ((!data || (data.length === 0 && !searchText)) && !authorizationInProgress) {
+        setTimeout(() => {
+          console.log("Reloading tasks");
+          updateTasks();
+        }, 500);
+      }
+    },
+  });
+  const {
+    data: projects,
+    isLoading: isLoadingProjects,
+    revalidate: updateProjects,
+  } = useCachedPromise(getProjects, ["", 1000], {
+    onData: (data) => {
+      if ((!data || data.length === 0) && !authorizationInProgress) {
+        setTimeout(() => {
+          console.log("Reloading projects");
+          updateProjects();
+        }, 500);
+      }
+      if (props.launchContext?.projectId) {
+        setProjectId(props.launchContext.projectId);
+      }
+    },
+  });
+
+  return (
+    <List
+      isLoading={isLoadingTasks}
+      throttle
+      pagination={pagination}
+      onSearchTextChange={setSearchText}
+      searchBarAccessory={
+        <List.Dropdown
+          isLoading={isLoadingProjects}
+          tooltip={"Filter by project"}
+          value={projectId}
+          onChange={(newValue) => setProjectId(newValue)}
+        >
+          <List.Dropdown.Item title="All" value="" key="all" />
+          {projects &&
+            Array.isArray(projects) &&
+            projects.map((project) => <List.Dropdown.Item title={project.name} value={project.id} key={project.id} />)}
+        </List.Dropdown>
+      }
+    >
+      {tasks &&
+        Array.isArray(tasks) &&
+        tasks.filter((value) => value.projectId === projectId).map((task) => <TaskItem key={task.id} task={task} />)}
```

Comment:
When I choose "All" in Dropdown I don't see any tasks - let's add a check so it doesn't filter when there is no projectId e.g.

```suggestion
      {tasks &&
        Array.isArray(tasks) &&
        tasks.filter((value) => !projectId || (value.projectId === projectId)).map((task) => <TaskItem key={task.id} task={task} />)}
```